doc/source/programs/gdal_vector_rasterize.rst

```diff
@@ -0,0 +1,214 @@
+
+    Advanced Options:
+    --oo, --open-option <KEY=VALUE>                          Open options [may be repeated]
+    --if, --input-format <INPUT-FORMAT>                      Input formats [may be repeated]
+
+    For more details, consult https://gdal.org/programs/gdal_vector_rasterize.html
+
+
+Description
+-----------
+
+:program:`gdal vector rasterize` burns vector geometries into a raster.
+
+The following options are available:
+
+Standard options
+++++++++++++++++
+
+.. include:: gdal_options/of_raster_create_copy.rst
+
+.. include:: gdal_options/co.rst
+
+.. option:: --band, -b <BAND>
+
+    The band(s) to burn values into. May be repeated.
+
+.. option:: --invert
+
+    Invert rasterization. Burn the fixed burn value, or the burn value associated with the first feature into all parts of the image not inside the provided polygon.
+
+    .. note::
+
+        When the vector features contain a polygon nested within another polygon (like an island in a lake), GDAL must be built against GEOS to get correct results.
+
+.. option:: --all-touched
+
+    Enables the ALL_TOUCHED rasterization option.
+
+    .. note::
+
+       When this option is enabled, the order of the input features (lines or polygons) can affect the results. When two features touch each other, the last one (i.e. topmost) will determine the burned pixel value at the edge. You may wish to use the `--sql` option to reorder the features (ORDER BY) to achieve a more predictable result.
+
+.. option:: --burn <BURN>
+
+    Burn value. May be repeated.
+
+.. option:: -a, --attribute-name <ATTRIBUTE-NAME>
+
+    Attribute name.
+
+.. option:: --3d
+
+    Indicates that a burn value should be extracted from the "Z" values of the feature. Works with points and lines (linear interpolation along each segment). For polygons, works properly only if the are flat (same Z value for all vertices).
+
+.. option:: --add
+
+   Instead of burning a new value, this adds the new value to the existing raster. Suitable for heatmaps for instance.
+
+.. option:: -l, --layer, --layer-name <LAYER-NAME>
+
+    Indicates the layer(s) from the datasource that will be used for input features. May be specified multiple times, but at least one layer name or a -sql option must be specified (not both).
+
+.. option:: --where <WHERE>
+
+    An optional SQL WHERE style query expression to be applied to select features to burn in from the input layer(s).
+
+.. option:: --sql <SQL>|@<filename>
+
+    An SQL statement to be evaluated against the datasource to produce a virtual layer of features to be burned in.
+    Starting with GDAL 3.7, the @filename syntax can be used to indicate that the content is in the pointed filename.
+
+.. option:: --dialect <DIALECT>
+
+    SQL dialect. In some cases can be used to use (unoptimized) OGR SQL instead of the native SQL of an RDBMS by passing OGRSQL. The "SQLITE" dialect can also be used with any datasource.
```

Comment:
https://gdal.org/en/latest/programs/gdal_vector_sql.html#cmdoption-dialect can be used to link to the OGR SQL and SQLITE dialect pages